discovery-infra/tests/test_validations.py

```diff
@@ -0,0 +1,76 @@
+import json
+import pytest
+
+from test_infra import consts
+from tests.base_test import BaseTest
+from tests.conftest import env_variables
+
+
+class TestValidations(BaseTest):
+    @pytest.mark.regression
+    def test_basic_cluster_validations(self, api_client, cluster):
+        cluster_id = cluster().id
+
+        # Check initial validations
+        self.wait_until_cluster_is_pending_for_input(cluster_id, api_client)
+        orig_cluster = api_client.cluster_get(cluster_id)
+        self.assert_cluster_validation(orig_cluster, "configuration", "pull-secret-set", "success")
+        self.assert_cluster_validation(orig_cluster, "hosts-data", "all-hosts-are-ready-to-install", "success")
+        self.assert_cluster_validation(orig_cluster, "hosts-data", "sufficient-masters-count", "failure")
+        self.assert_cluster_validation(orig_cluster, "network", "machine-cidr-defined", "failure")
+        self.assert_cluster_validation(orig_cluster, "network", "machine-cidr-equals-to-calculated-cidr", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "api-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "api-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "dns-domain-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "cluster-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "service-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "no-cidrs-overlapping", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "network-prefix-valid", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "ntp-server-configured", "success")
+
+        # Check base DNS domain
+        api_client.update_cluster(cluster_id, {"base_dns_domain": ""})
```

Comment:
update empty dns doesn't fail?

Reply:
It's not a required parameter when creating the cluster, so the user should be allowed to unset the value

Reply:
but we do have a validation for dns, did the test passed?

Reply:
We have validation, if it's non-empty.  Yes, the test passes.